_pages/documentation/general_docs/kconfig_build_system.md

```diff
@@ -0,0 +1,198 @@
+---
+layout: documentation
+title: Kconfig Build System
+doc: gem5 documentation
+parent: kconfig_build_system
+permalink: /documentation/general_docs/kconfig_build_system/
+---
+
+For advanced users who need to build gem5 with multiple ISAs or customized the
+build options, such as ruby memory protocol. Users need to familiar with
+Kconfig system.
+
+## Build GEM5 with Kconfig Build System
+
+```bash
+scons [OPTIONS] Kconfig_command TARGET
+```
+
+Supported kconfig commands include
+
+- defconfig
+- setconfig
+- menuconfig
+- guiconfig
+- listnewconfig
+- oldconfig
+- olddefconfig
+- savedefconfig
+
+The most common use of options are `defconfig`, `setconfig` and `menuconfig`.
+
+Use `scons --help` to list these commands, and for additional information.
+
+To build gem5 with kconfig, there are now two steps.
+
+Initial configuration
+
+```bash
+scons defconfig gem5_build build_opts/ALL
+```
+
+This will create a configuration for the gem5_build build directory based on
+`build_opts/ALL`. The configuration is stored within the build directory.
+
+To build a target in a configured build directory, run a scons command similar
+to before.
+
+```bash
+scons gem5_build/gem5.opt
+```
+
+To build gem5 kconfig with customized kconfig options, the additional step
+is required between **initial configuration** and **building the target**
+
+- Directly set kconfig options
+
+```bash
+scons setconfig gem5_build USE_KVM=y
+```
+
+- set kconfig options by `menuconfig`
+
+```bash
+scons menuconfig gem5_build
+```
+
+## Details of Kconfig Commands
+
+### defconfig
+
+Set up a config using values specified in a defconfig file, or if no value is
+given, use the default. The second argument specifies the defconfig file. All
+of available gem5 defconfig files are in `build_opts` directory.
+
+A defconfig file in the build_opts directory can be implicitly specified in
+the build path via `build/<defconfig file>/`
+
+The user can also specify the defconfig FILE outside the gem5 repository.
+
+```bash
+scons defconfig gem5_build build_opts/RISCV
+```
+
+### setconfig
+
+Set values in an existing config directory as specified on the command line.
```

Comment:
How would one know what variables are available to set? I don't really understand the use case here.

Reply:
Please use `menuconfig` and `guiconfig` to list all of config options

Reply:
This information should be in the documenation.

Apologies if my prior review wasn't clear... any "questions" I posed were because they are questions users/developers might ask that are not clear in the documentation.

When writing documentation for 100s-1000s or even 10,000s of people you have to be very clear and precise. You may also have to answer a lot of obvious (to you) questions :smile:

Reply:
Added in the doc